packages/ember-glimmer/lib/environment.js

```diff
@@ -72,7 +73,13 @@ export default class Environment extends GlimmerEnvironment {
       }
     }
 
-    return super.refineStatement(statement);
+    let nativeSyntax = super.refineStatement(statement);
+
+    if (!nativeSyntax && key && this.hasHelper(key)) {
```

Comment:
yea I thought about that, but textStatements (errthing false/undefined) were also being passed into `refineStatement()`, which was problematic (especially when `key` is undefined). Anyway I updated the logic and tests pass